index.ts

```diff
@@ -0,0 +1,245 @@
+    const releaseLock = await mutex.acquire()
+    return releaseLock
+  }
+
+  _lookupMutex (lockId: string): any {
+    let mutex = this.lockMap[lockId]
+    if (!mutex) {
+      mutex = new Mutex()
+      this.lockMap[lockId] = mutex
+    }
+    return mutex
+  }
+
+  /**
+   * Function returns the nonce details from teh network based on the latest block 
+   * and eth_getTransactionCount method
+   * 
+   * @param address the hex string for the address whose nonce we are calculating
+   * @returns {Promise<NetWorkNextNonce>}
+   */
+  async _getNetworkNextNonce (address: string): Promise<NetWorkNextNonce> {
+    // calculate next nonce
+    // we need to make sure our base count
+    // and pending count are from the same block
+    const blockNumber: Promise<any> = await this.blockTracker.getLatestBlock()
+    const baseCountBN = await this.ethQuery.getTransactionCount(address, blockNumber)
+    const baseCount:number = baseCountBN.toNumber()
+    assert(Number.isInteger(baseCount), `nonce-tracker - baseCount is not an integer - got: (${typeof baseCount}) "${baseCount}"`)
+    const nonceDetails = { blockNumber, baseCount }
+    return { name: 'network', nonce: baseCount, details: nonceDetails }
+  }
+
+  /**
+   * Function returns the highest of the confirmed transaction from the address.
+   * @param address the hex string for the address whose nonce we are calculating
+   */
+  _getHighestLocallyConfirmed (address:string): number {
+    const confirmedTransactions: Transactions[] = this.getConfirmedTransactions(address)
+    const highest = this._getHighestNonce(confirmedTransactions)
+    return Number.isInteger(highest) ? highest + 1 : 0
+  }
+
+  /**
+   * Function returns highest nonce value from the transcation list provided
+   * @param txList list of transactions
+   */
+  _getHighestNonce (txList:Transactions[] ): number {
+    const nonces = txList.map((txMeta) => {
+      const nonce = txMeta.txParams.nonce
+      assert(typeof nonce == 'string', 'nonces should be hex strings')
+      return parseInt(nonce, 16)
+    })
+    const highestNonce = Math.max.apply(null, nonces)
+    return highestNonce
+  }
+
+ 
+  /**
+    * Function return the nonce value higher than the highest nonce value from the transaction list
+    *  starting from startPoint
+    * @param txList {array} - list of txMeta's
+    * @param startPoint {number} - the highest known locally confirmed nonce
+    */
+  _getHighestContinuousFrom (txList: Transactions[], startPoint:number): HighestContinuousFrom {
+    const nonces = txList.map((txMeta) => {
+      const nonce = txMeta.txParams.nonce
+      assert(typeof nonce == 'string', 'nonces should be hex strings')
+      return parseInt(nonce, 16)
+    })
+
+    let highest = startPoint
+    while (nonces.includes(highest)) {
+      highest++
+    }
+
+    return { name: 'local', nonce: highest, details: { startPoint, highest } }
+  }
+
+}
+module.exports = NonceTracker
```

Comment:
After you've implemented the suggested changes for `tsconfig.json`, you can achieve the desired effect by doing this:
```suggestion
export = NonceTracker
```